modules/bm_sim/include/bm_sim/switch.h

```diff
@@ -551,6 +552,10 @@ class SwitchWContexts : public DevMgr, public RuntimeInterface {
     return contexts.at(cxt_id).add_component<T>(ptr);
   }
 
+  void set_lookup_factory(LookupStructureFactory * new_factory) {
```

Comment:
let's make this a smart pointer (shared or unique), since this is a client-facing API.